umh-core/pkg/communicator/actions/edit-dataflowcomponent.go

```diff
@@ -394,17 +394,60 @@ func (a *EditDataflowComponentAction) Execute() (interface{}, map[string]interfa
 		// Convert each processor configuration in the pipeline
 		processors := []interface{}{}
 
-		for processorName, processor := range a.payload.Pipeline {
-			var procConfig map[string]interface{}
-			err := yaml.Unmarshal([]byte(processor.Data), &procConfig)
+		// Check if we have numeric keys (0, 1, 2, ...) and use them to preserve order
+		numericKeys := make(map[int]string)
+		hasNumericKeys := true
+
+		// Try to parse all keys as integers
+		for processorName := range a.payload.Pipeline {
```

Comment:
if you do a re-usable function for deploying, you can here re-use that one.

but don't create a helper.go or utils.go, that is an antipatter. use a describptive name, e.g., frontend_processor_parsing.go or somethign